pyro/infer/reparam/strategies.py

```diff
@@ -0,0 +1,180 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+These reparametrization strategies are registered with
+:func:`~pyro.poutine.reparam_messenger.register_reparam_strategy` and are
+accessed by name via ``poutine.reparam(config=name_of_strategy)`` .
+See :func:`~pyro.poutine.handlers.reparam` for usage.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Callable, Dict, Optional, Union
+
+import torch
+
+import pyro.distributions as dist
+import pyro.poutine as poutine
+
+from .loc_scale import LocScaleReparam
+from .projected_normal import ProjectedNormalReparam
+from .reparam import Reparam
+from .softmax import GumbelSoftmaxReparam
+from .stable import LatentStableReparam, StableReparam, SymmetricStableReparam
+from .transform import TransformReparam
+
+
+class Strategy(ABC):
```

Comment:
should there be a `prefix` arg like in `AutoGuides` to avoid possible name mangling... i guess this would require changes on the `Reparam` end...

Reply:
I guess we could add an `infix` arg to separate user-provided names from reparam suffixes, but I think that would best live at the level of base `Reparam` class and seems outside the scope of this PR. And maybe we should wait until we have at least one report of real-world name conflict before adding this bit of complexity.

Reply:
sounds reasonable but should we add a note in the docs to the effect of 

"Note that reparametrizers introduce auxiliary sites with names like `my_latent_decentered`. In case of name conflict you should rename your sample sites accordingly."

Reply:
SGTM, I'll address in a separate PR. I think we may be able to add some error checking as well.